tasks/src/main/scala/dagr/tasks/misc/BwaMemStreamed.scala

```diff
@@ -53,48 +54,25 @@ class BwaMemStreamed(unmappedBam: PathToBam,
                      mergeBamAlignmentMem: String = "2G",
                      fifoBufferMem: String = "512M",
                      processAltMappings : Boolean = true
-                    ) extends PipedTask with VariableResources {
-  this.name = "BwaMemStreamed"
-  override protected val tasks = ListBuffer[ProcessTask]()
+                    ) extends Pipeline {
 
-  tasks ++= SamToFastq(in = unmappedBam, out = Io.StdOut).requires(memory = Memory(samToFastqMem)).getTasks
-  tasks ++= new FifoBuffer().requires(Cores(0), Memory(fifoBufferMem)).getTasks
-  tasks ++= new BwaMem(fastq = Io.StdIn, ref = ref, minThreads = minThreads, maxThreads = maxThreads, memory = Memory(bwaMemMemory)).getTasks
-  tasks ++= new FifoBuffer().requires(memory = Memory(fifoBufferMem)).getTasks
+  override def build(): Unit = {
+    val samToFastq = SamToFastq(in=unmappedBam, out=Io.StdOut).requires(Cores(0.5), Memory(samToFastqMem))
+    val fqBuffer   = new FifoBuffer[Fastq].requires(memory=Memory(fifoBufferMem))
+    val bwaMem     = new BwaMem(fastq = Io.StdIn, ref = ref, minThreads = minThreads, maxThreads = maxThreads, memory = Memory(bwaMemMemory))
+    val bwaBuffer  = new FifoBuffer[Sam].requires(memory=Memory(fifoBufferMem))
 
-  // Optionally tie in alt-processing if requested and the .alt file exists
-  {
+    var pipe = samToFastq | fqBuffer | bwaMem | bwaBuffer
+
+    // Optionally tie in alt-processing if requested and the .alt file exists
     val alt = PathUtil.pathTo(ref + ".alt")
     if (processAltMappings && alt.toFile.exists()) {
-      tasks ++= new BwaK8AltProcessor(altFile = alt).getTasks
-      tasks ++= new FifoBuffer().requires(memory = Memory(fifoBufferMem)).getTasks
+      val k8       = new BwaK8AltProcessor(altFile = alt)
+      val k8Buffer = new FifoBuffer[Sam].requires(memory=Memory(fifoBufferMem))
+      pipe = pipe | k8 | k8Buffer
     }
```

Comment:
It would be neat to be able to do something like this:

```
override def build(): Unit = {
    val alt = PathUtil.pathTo(ref + ".alt")

    val samToFastq = SamToFastq(in=unmappedBam, out=Io.StdOut).requires(Cores(0.5), Memory(samToFastqMem))
    val fqBuffer   = new FifoBuffer[Fastq].requires(memory=Memory(fifoBufferMem))
    val bwaMem     = new BwaMem(fastq = Io.StdIn, ref = ref, minThreads = minThreads, maxThreads = maxThreads, memory = Memory(bwaMemMemory))
    val bwaBuffer  = new FifoBuffer[Sam].requires(memory=Memory(fifoBufferMem))

    // Optionally tie in alt-processing if requested and the .alt file exists
    var pipe = samToFastq | fqBuffer | bwaMem | bwaBuffer | if (processAltMappings && alt.toFile.exists()) {
      val k8       = new BwaK8AltProcessor(altFile = alt)
      val k8Buffer = new FifoBuffer[Sam].requires(memory=Memory(fifoBufferMem))
      k8 | k8Buffer
    } 
    else {
      PipeEnd
    }

    val mergeBam = new MergeBamAlignment(unmapped=unmappedBam, mapped=Io.StdIn, out=mappedBam, ref=ref).requires(memory=Memory(mergeBamAlignmentMem))
    root ==> (pipe | mergeBam).withName("BwaMemThroughMergeBam")
  }
```

I am drawing inspiration from the spray-routing DSL (ex. `PathEnd`)